modules/microshift-nw-router-config-IPs.adoc

```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+// * microshift_networking/microshift-nw-router.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-config-ip-addresses_{context}"]
+= Configuring router IP addresses
+
+You can restrict the network traffic to the router by configuring specific IP addresses. For example:
+* Use cases where the router is reachable only on internal networks, but not on northbound public networks.
+* Use cases where the router is reachable only by northbound public networks, but not on internal networks.
+* Use cases where the router is reachable by both internal networks and northbound public networks, but on separate IP addresses.
+
+.Prerequisites
+
+* You have created a {microshift-short} `config.yaml` file.
+
+* The {oc-first} is installed.
+
+//Q: Any other prereqs?
+
+.Procedure
+
+. Update the IP addresses in the `apiServer.advertiseAddress` field in the {microshift-short} `config.yaml` as shown in the following example:
++
+.Example `config.yaml` router settings
+[source,yaml]
+----
+apiServer:
+  advertiseAddress: 10.44.0.0/32
+#....
+ingress:
+  listenAddress: # <1>
+    - "" # <2>
+  ports:
+    http: 80 # <3>
+    https: 443 # <4>
+  routeAdmissionPolicy:
+    namespaceOwnership: InterNamespaceAllowed <5>
+  status: Managed # <6>
+# ....
+----
+<1> Defaults to the entire network of the host. Valid values are IP addresses or network interface names (NICs) as configured in the host.
+<2> Valid values are a single IP address or a list of IP addresses.
+<3> Valid value is a single port in the 1-65535 range. Cannot be the same as the value of the `ports.https` field.
+<4> Valid value is a single port in the 1-65535 range. Cannot be the same as the value of the `ports.http` field.
+<5> Default value. Allows routes to claim different paths of the same host name across namespaces.
+<6> Default value. `Managed` is required for the ingress ports to remain open.
++
+[TIP]
+====
+If you complete all the configurations that you need to make in the {microshift-short} `config.yaml` file at the same time, you can minimize system restarts.
+====
+
+//.Verification
+//TODO Yes, Pablo to provide example; see end-to-end tests
```

Comment:
Let me prepare an example that is concise enough for the docs using an internal IP.

Reply:
1. debug node and get the host's ip addresses and the NICs
```
% oc debug node/dhcp-1-235-168.arm.eng.rdu2.redhat.com  
Starting pod/dhcp-1-235-168armengrdu2redhatcom-debug-tmvxm ...
To use host binaries, run `chroot /host`
Pod IP: 10.1.235.168
If you don't see a command prompt, try pressing enter.
sh-4.4# chroot /host
sh-5.1# ip address | grep "inet "
    inet 127.0.0.1/8 scope host lo
    inet 10.1.235.168/24 brd 10.1.235.255 scope global dynamic noprefixroute ens3
    inet 192.168.124.153/24 brd 192.168.124.255 scope global dynamic noprefixroute ens4
    inet 10.44.0.0/32 scope global br-ex
    inet 169.254.169.2/29 brd 169.254.169.7 scope global br-ex
    inet 10.42.0.2/24 brd 10.42.0.255 scope global ovn-k8s-mp0
sh-5.1#
```
2. create the /etc/microshift/config.yaml file under the debug node, use 10.1.235.168 and NIC ens4 for example.
```
sh-5.1# cat /etc/microshift/config.yaml
ingress:
    listenAddress:
        - "10.1.235.168"
        - "ens4"
```

Reply:
@ShudiLi If we just configured the yaml, it's circular for a user to re-create it, right? Wonder if a user would really do this in a production environment?